content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md

```diff
@@ -152,6 +152,17 @@ Pick a control plane node that you wish to upgrade first. It must have the `/etc
    [upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets if you haven't already done so.
    ```
 
+   {{< note >}}
+   Before v1.27 (including v1.27), kubeadm will upgrade the addons (including CoreDNS and kube-proxy) immediately during 
+   `kubeadm upgrade apply`, regardless of whether there are other control plane instances that have not been upgraded, which 
+   may cause compatibility problems. Since v1.28, kubeadm will always checks whether all the control plane instances have 
+   been upgraded before starting to upgrade the addons. You **MUST** perform control plane nodes upgrade sequentially, and 
+   the addons will be upgraded after the last control plane has been upgraded.
+   If you want to keep the old upgrade behavior, please enable `UpgradeAddonsBeforeControlPlane` feature gate by 
+   `kubeadm upgrade apply --feature-gates=UpgradeAddonsBeforeControlPlane=true`. 
+   But please note that this feature gate has been **deprecated** and will be removed in a future release.
```

Comment:
```suggestion
   The Kubernetes project does not in general recommend enabling this feature gate (you should instead
   change your upgrade process or cluster addons so that you do not need to enable the legacy behavior.
   The `UpgradeAddonsBeforeControlPlane` feature gate will be removed in a future release.
```